Implement circuit interface

lomination opened the pull request.

Nota-yob commented:
You should avoid using the old input system (defining the input inside the code) but instead use the GUI input system and make callback function that get executed when the corresponding input is triggered by the key.

lomination commented:
> You should avoid using the old input system (defining the input inside the code) but instead use the GUI input system and make callback function that get executed when the corresponding input is triggered by the key.

I will take a look at this, thanks

lomination commented:
> [!NOTE]
> The last commit is not working but you can try the previous one. Controls are 'R' to interract, 'N' to show the interaction range, and 'I' to see how many interactable objects are in the range.

I would like to discuss about the implementation and above all the user interface of the interaction with the circuit interface. I would like to see that with you @Nota-yob . We could also have a look at the input management together if you accept to.

Nota-yob commented:
Tomorrow is OK for you ? @lomination

lomination commented:
At what time?

lomination commented:
I added a second interactable sphere and fixes the conflicts issues: the nearest interactable is used when the interaction key is pressed.

I also made the game ungrab the cursor but the camera is still following the cursor movements.

The code should be easy and documented enough for anyone to understand (if not, please let me know).

Nota-yob commented:
I added a reference to the Cinemachine Free Look component that permits us to disable/enable the input and i toggles it when the object is interracted. I also modified your ShowCursor method to become ToggleCursor and removed the parameter because u used the private property that can be accessd in this method.

lomination commented:
I basically made the interactable objects in range glow. The nearest one from the player glows more.

The glow effect has been made using the Outline script from the added package Quick Outline (made for this purpose).

This feature implied to change the IInteractable interface to an abstract class so it can access the GetComponent (in this case `GetComponent<Outline>()`) method. A few methods have been added but no major change and the code should be clear, short and documented enough for anyone to understand.

lomination commented:
Since it is really work in progress, I won't provide explanations for now. They will come as soon as the idea becomes more concrete.

lomination commented:
I'm sorry for this early merge... I did not plan to.

I implemented a breadboard on which you can move dipoles. These dipoles can be dragged and rotated (pressing R). I also implemented wires that can be created by holding the mouse down between two nodes. To delete a wire, you can click on it or try to create a wire above another one. The latter method will make the breadboard enter in '_onWireDeletion', i. e. no wire can be created anymore and every encountered wire will be destroyed. To exit this mode, simply release the mouse button. Please feel free to try and give feedback on the interface.

I'm gonna opens issues to track the work that still needs to be done.
